app/models/team.rb

```diff
@@ -65,15 +88,32 @@ def add_member(user, _assignment_id = nil)
     can_add_member = false
     unless full?
       can_add_member = true
-      t_user = TeamsUser.create(user_id: user.id, team_id: id)
       parent = TeamNode.find_by(node_object_id: id)
-      TeamUserNode.create(parent_id: parent.id, node_object_id: t_user.id)
       add_participant(parent_id, user)
+      participant = AssignmentParticipant.find_by(parent_id: parent_id, user_id: user.id)
+      t_user = TeamsUser.create(participant_id: participant.id, team_id: id)
+      TeamUserNode.create(parent_id: parent.id, node_object_id: t_user.id)
```

Comment:
Can you let me know what TeamUserNode does? How can we use participant_id here?

Reply:
To display in the team in tree structure, TeamUserNode is created with id from teams_users table